src/OrleansProviders/Streams/Memory/MemoryBatchContainer.cs

```diff
@@ -9,10 +7,10 @@
 
 namespace Orleans.Providers
 {
-    [Serializable]
     internal class MemoryBatchContainer<TSerializer> : IBatchContainer
```

Comment:
I'll add the [Serializable] attribute back. Note that [Serializable] is an attribute which hints to us that a serializer should be generated for some type, but also gives permission to BinaryFormatter to serialize that type. I want to avoid the latter wherever possible, since we don't have enough control over it so that we can inject context in cases like this - where we're serializing services instead of just POCOs

> We've not only made the serialization manager no longer global, we've made it the application layer's problem to figure out how to get it, instead of passing it to systems that need it.

The answer is that they should be constructed by DI. The only types which should be instantiated via `new` are POCOs and types which aren't shared (via serialization or otherwise)